Source/WebCore/platform/network/NetworkStorageSession.cpp

```diff
@@ -159,13 +159,16 @@ bool NetworkStorageSession::shouldBlockCookies(const URL& firstPartyForCookies,
         return true;
     case ThirdPartyCookieBlockingMode::AllExceptBetweenAppBoundDomains:
         return !shouldExemptDomainPairFromThirdPartyCookieBlocking(firstPartyDomain, resourceDomain);
+    case ThirdPartyCookieBlockingMode::AllExceptManagedDomains:
+        return !m_managedDomains.contains(firstPartyDomain) || !hasHadUserInteractionAsFirstParty(firstPartyDomain);
```

Comment:
This is complex to understand. I suggest a helper variable and flipping the OR to an AND. Like this:
```
bool isManagedDomainWithUserInteraction = m_managedDomains.contains(firstPartyDomain) && hasHadUserInteractionAsFirstParty(firstPartyDomain);
return !isManagedDomainWithUserInteraction;
```